### PR TITLE
Add DSL for quality pipeline

### PR DESCRIPTION
**Pull request by michaelyoungstrom**
DSL for master and pr quality pipeline jobs.

(Will be changing the branch and refs in the master file. Those are temp branches for testing).

### CODE REVIEW

File: platform/jobs/pipelines/edxPlatformQualityMaster.groovy [hunk 1]
```diff
@@ -0,0 +1,46 @@
+package platform
+
+import org.yaml.snakeyaml.Yaml
+import static org.edx.jenkins.dsl.JenkinsPublicConstants.JENKINS_PUBLIC_LOG_ROTATOR
+import static org.edx.jenkins.dsl.JenkinsPublicConstants.JENKINS_PUBLIC_TEAM_SECURITY
```
Comment (jmbowman) [inline platform/jobs/pipelines/edxPlatformQualityMaster.groovy]: Are the first and third imports here actually needed?

File: platform/views/platformViews.groovy [hunk 2]
```diff
@@ -56,3 +56,12 @@ listView("edx-platform-pipeline-pr-tests") {
     }
     columns DEFAULT_VIEW.call()
 }
+
+listView("edx-platform-pipeline-master-tests") {
+    description('Pipeline Jobs for running tests pushes to master branch of the edx-platform')
```
Comment (jmbowman) [inline platform/views/platformViews.groovy]: "tests pushes" -> "tests on pushes"?
"of the edx-platform" -> "of edx-platform"?

File: platform/jobs/pipelines/edxPlatformQualityPipelinePr.groovy [hunk 3]
```diff
@@ -0,0 +1,80 @@
+package platform
+
+import org.yaml.snakeyaml.Yaml
+import static org.edx.jenkins.dsl.JenkinsPublicConstants.JENKINS_PUBLIC_LOG_ROTATOR
+import static org.edx.jenkins.dsl.JenkinsPublicConstants.JENKINS_PUBLIC_TEAM_SECURITY
+import static org.edx.jenkins.dsl.JenkinsPublicConstants.JENKINS_PUBLIC_LOG_ROTATOR
```
Comment (jmbowman) [inline platform/jobs/pipelines/edxPlatformQualityPipelinePr.groovy]: This last import is a duplicate, and the one before it seems unused.

File: platform/jobs/pipelines/edxPlatformLettucePipelinePr.groovy [hunk 4]
```diff
@@ -2,7 +2,6 @@ package platform
 
 import org.yaml.snakeyaml.Yaml
 import static org.edx.jenkins.dsl.JenkinsPublicConstants.JENKINS_PUBLIC_LOG_ROTATOR
-import static org.edx.jenkins.dsl.JenkinsPublicConstants.JENKINS_PUBLIC_TEAM_SECURITY
```
Comment (estute) [inline platform/jobs/pipelines/edxPlatformLettucePipelinePr.groovy]: I know this was copy/paste, but I am just curious what your thoughts are for private pipeline jobs.We can talk about that in slack if you want

File: platform/jobs/pipelines/edxPlatformQualityPipelinePr.groovy [hunk 5]
```diff
@@ -0,0 +1,78 @@
+package platform
+
+import org.yaml.snakeyaml.Yaml
+import static org.edx.jenkins.dsl.JenkinsPublicConstants.JENKINS_PUBLIC_LOG_ROTATOR
+
+/* stdout logger */
+Map config = [:]
+Binding bindings = getBinding()
+config.putAll(bindings.getVariables())
+PrintStream out = config['out']
```
Comment (estute) [inline platform/jobs/pipelines/edxPlatformQualityPipelinePr.groovy]: ^ this is only used for creating a logger to make debugging easier when parsing yml files

Comment (estute) [reply platform/jobs/pipelines/edxPlatformQualityPipelinePr.groovy]: nevermind

File: platform/jobs/pipelines/edxPlatformLettucePipelinePr.groovy (hunk 4 above)
Comment (michaelyoungstrom) [reply platform/jobs/pipelines/edxPlatformLettucePipelinePr.groovy]: Perhaps we should add a separate ticket to incorporate private jobs

Comment (estute) [reply platform/jobs/pipelines/edxPlatformLettucePipelinePr.groovy]: Sounds good

**Pull request closed without merging**